app/src/main/java/com/sun/hero_01/ui/champion/ChampionFragment.kt

```diff
@@ -1,22 +1,69 @@
 package com.sun.hero_01.ui.champion
 
+import android.content.Context
+import android.content.SharedPreferences
 import android.os.Bundle
 import android.view.View
+import android.widget.PopupMenu
 import android.widget.Toast
 import com.sun.hero_01.R
 import com.sun.hero_01.base.BaseFragment
 import com.sun.hero_01.data.model.Hero
 import com.sun.hero_01.data.source.HeroRepository
 import com.sun.hero_01.data.source.remote.HeroRemoteDataSource
+import com.sun.hero_01.ui.compare.CompareFragment
 import com.sun.hero_01.ui.detail.DetailFragment
 import com.sun.hero_01.ui.search.SearchFragment
 import com.sun.hero_01.utils.OnItemRecyclerViewListener
 import com.sun.hero_01.utils.extensions.replaceFragment
 import kotlinx.android.synthetic.main.fragment_champion.*
+import kotlinx.android.synthetic.main.item_layout_hero.*
 
 class ChampionFragment : BaseFragment(), ChampionContract.View, OnItemRecyclerViewListener<Hero> {
 
-    private val adapter by lazy { ChampionAdapter(this) }
+    private val adapterChampion by lazy {
+        ChampionAdapter(this) { idName, view ->
+            onShowMenu(idName, view)
+        }
+    }
+
+    private val sharedKey = "kotlin shared"
```

Comment:
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md#210-string-constants-naming-and-values